python/ray/tune/tests/test_api.py

```diff
@@ -364,69 +363,59 @@ def train(config, reporter):
         )
 
     def testLogdir(self):
-        def train(config, reporter):
-            assert (
-                os.path.join(ray._private.utils.get_user_temp_dir(), "logdir", "foo")
-                in os.getcwd()
-            ), os.getcwd()
-            reporter(timesteps_total=1)
-
-        register_trainable("f1", train)
-        run_experiments(
-            {
-                "foo": {
-                    "run": "f1",
-                    "local_dir": os.path.join(
-                        ray._private.utils.get_user_temp_dir(), "logdir"
-                    ),
-                    "config": {"a": "b"},
-                }
-            }
-        )
+        logdir = os.path.join(ray._private.utils.get_user_temp_dir(), "logdir")
+
+        def train_fn(config):
+            assert logdir in os.getcwd(), os.getcwd()
+            train.report(dict(timesteps_total=1))
+
+        register_trainable("f1", train_fn)
+        with unittest.mock.patch.dict(os.environ, {"TEST_TMPDIR": logdir}):
```

Comment:
Should we be using the "public" API for this test - `RAY_AIR_LOCAL_CACHE_DIR`?

Reply:
SG, replaced